<a href="https://colab.research.google.com/github/Akshay069/Yes-bank-stock-closing-price-prediction/blob/main/Yes_Bank_Closing_price_predictions.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Problem Statement**

---


Yes Bank is a well-known bank in the Indian financial domain. Since 2018, it has been in the
news because of the fraud case involving Rana Kapoor. Owing to this fact, it was interesting
to see how that impacted the stock prices of the company and whether Time series models
or any other predictive models can do justice to such situations. This dataset has monthly
stock prices of the bank since its inception and includes closing, starting, highest, and lowest
stock prices of every month. The main objective is to predict the stock’s closing price of the
month.



###**Data Overview**

---
Fisrt things first\
Data is nothing but a information so,
I will first load data into Pandas DataFrame and take a tour of our data on the basis of which we will take decisions.


In [3]:
# Lets call out some important liabraries which i will going to use un this project
import numpy as np
import pandas as pd

In [2]:
# Lets mount google drive.
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [4]:
# Lets call data using Pandas DataFrame.
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/AlmaBetter/Machine Learning/Supervised Learning/Capston Project/Copy of data_YesBank_StockPrices.csv')

In [7]:
# Shape of dataset.
print('* Number of obervations are', df.shape[0])
print('* Number of features are', df.shape[1])

* Number of obervations are 185
* Number of features are 5


In [10]:
# Lets take a first look at our data.
df.head()

Date   Open   High    Low  Close
0  Jul-05  13.00  14.00  11.25  12.46
1  Aug-05  12.58  14.88  12.55  13.42
2  Sep-05  13.48  14.87  12.27  13.30
3  Oct-05  13.20  14.47  12.40  12.99
4  Nov-05  13.35  13.88  12.88  13.41

As we see above there are only five features in our dataset and only 185 observations. we will see the descriptions about all the features below on by one.


**Date:** It denotes the month & year with respect to the price of the stock.

**Open:** The price at which a stock started trading that month.

**High:** Refers to the maximum price of the stock.

**Low:** Refers to the minimum price of the stock.

**Close:** Refers to the final trading price for that month.

**'Close' is the dependent variable in this case as we are going to predict the closing price of the stock**

In [12]:
# Lets have a look at overall summary of data.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 185 entries, 0 to 184
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Date    185 non-null    object 
 1   Open    185 non-null    float64
 2   High    185 non-null    float64
 3   Low     185 non-null    float64
 4   Close   185 non-null    float64
dtypes: float64(4), object(1)
memory usage: 7.4+ KB


From above we can see that there is no null value present in our dataset.\
also the dtype(data type) of 'date' coulmn is in object so we have to convert it into datetime fromat  

In [21]:
# before doing any manipulation to our dataset, Lets make a copy of it in order to keep original dataset intact.
df_copy = df.copy()

In [15]:
df_copy['Date'][0]

'Jul-05'

We can see that date contain string values in it. Lets convert it into datetime format

In [22]:
from datetime import datetime
df_copy['Date'] = df_copy['Date'].apply(lambda x: datetime.strptime(x, '%b-%y'))

In [23]:
df_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 185 entries, 0 to 184
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   Date    185 non-null    datetime64[ns]
 1   Open    185 non-null    float64       
 2   High    185 non-null    float64       
 3   Low     185 non-null    float64       
 4   Close   185 non-null    float64       
dtypes: datetime64[ns](1), float64(4)
memory usage: 7.4 KB


We can see above, we have successfully converted date into datetime format

In [26]:
# Lets see if any duplicated values are there in our dataset.
df_copy.duplicated().sum()

0

We can see that we don't have any duplicate values in our dataset.

In [31]:
# Descriptive summary 
round(df_copy.describe(),2)

Open    High     Low   Close
count  185.00  185.00  185.00  185.00
mean   105.54  116.10   94.95  105.20
std     98.88  106.33   91.22   98.58
min     10.00   11.24    5.55    9.98
25%     33.80   36.14   28.51   33.45
50%     62.98   72.55   58.00   62.54
75%    153.00  169.19  138.35  153.30
max    369.95  404.00  345.50  367.90

There is a high difference between mean and median values in each column which indicates that the dataset is skewed.